features/lorawan/lorastack/phy/LoRaPHYIN865.h

```diff
@@ -54,308 +53,31 @@ class LoRaPHYIN865 : public LoRaPHY {
-     */
-    virtual int8_t get_alternate_DR(AlternateDrParams_t* alternateDr );
-
-    /*!
-     * \brief Calculates the back-off time.
-     *
-     * \param [in] calcBackOff A pointer to the function parameters.
-     */
-    virtual void calculate_backoff(CalcBackOffParams_t* calcBackOff );
-
-    /*!
-     * \brief Searches and sets the next random available channel.
-     *
-     * \param [in]  nextChanParams The parameters for the next channel.
-     *
-     * \param [out] channel The next channel to use for TX.
-     *
-     * \param [out] time The time to wait for the next transmission according to the duty cycle.
-     *
-     * \param [out] aggregatedTimeOff Updates the aggregated time off.
-     *
-     * \retval The function status [1: OK, 0: Unable to find a channel on the current datarate].
-     */
-    virtual bool set_next_channel(NextChanParams_t* nextChanParams,
-                                   uint8_t* channel, TimerTime_t* time,
-                                   TimerTime_t* aggregatedTimeOff );
-
-    /*!
-     * \brief Adds a channel.
-     *
-     * \param [in] channelAdd A pPointer to the function parameters.
-     *
-     * \retval The status of the operation.
-     */
-    virtual LoRaMacStatus_t add_channel(ChannelAddParams_t* channelAdd );
-
-    /*!
-     * \brief Removes a channel.
-     *
-     * \param [in] channelRemove A pointer to the function parameters.
-     *
-     * \retval True, if the channel was removed successfully.
-     */
-    virtual bool remove_channel(ChannelRemoveParams_t* channelRemove );
-
-    /*!
-     * \brief Sets the radio into continuous wave mode.
-     *
-     * \param [in] continuousWave A pointer to the function parameters.
-     */
-    virtual void set_tx_cont_mode(ContinuousWaveParams_t* continuousWave );
-
-    /*!
-     * \brief Computes a new datarate according to the given offset.
-     *
-     * \param [in] downlinkDwellTime The downlink dwell time configuration. 0: No limit, 1: 400ms
-     *
-     * \param [in] dr The current datarate.
-     *
-     * \param [in] drOffset The offset to be applied.
-     *
-     * \retval newDr The computed datarate.
-     */
-    virtual uint8_t apply_DR_offset(uint8_t downlinkDwellTime, int8_t dr, int8_t drOffset );
+    virtual uint8_t apply_DR_offset(int8_t dr, int8_t dr_offset );
 
 private:
-    uint8_t CountNbOfEnabledChannels( bool joined, uint8_t datarate, uint16_t* channelsMask, ChannelParams_t* channels, Band_t* bands, uint8_t* enabledChannels, uint8_t* delayTx );
-
-    // Global attributes
     /*!
      * LoRaMAC channels
      */
-    ChannelParams_t Channels[IN865_MAX_NB_CHANNELS];
+    channel_params_t channels[IN865_MAX_NB_CHANNELS];
 
     /*!
      * LoRaMac bands
      */
-    Band_t Bands[IN865_MAX_NB_BANDS];
+    band_t bands[IN865_MAX_NB_BANDS];
 
     /*!
      * LoRaMac channels mask
      */
-    uint16_t ChannelsMask[IN865_CHANNELS_MASK_SIZE];
+    uint16_t channel_masks[IN865_CHANNELS_MASK_SIZE];
```

Comment:
"mask" should be singular

Reply:
There can be multiple channel masks masking disparate blocks of 16 .

Reply:
But isn't that just one big mask? Is there any logical structure to it? The data type just looks like a big bitfield formed out of 16-bit processing units. Are the 16-bit boundaries significant?

Reply:
Yes, checked the core phy code - it thinks it's just one big bit mask for things that have like 72 channels - have as many 16-bit containers as needed for number of channels in the phy.

The code in the higher layers is not actually processing the mask according to that, just accessing `channel_masks[0]`.

Reply:
Yes that is true. In the parent class (LoRaPHY), some of the methods use channel_mask[0] as those methods make sense for PHY layer implementations who have 16 channels only. Those with more than 16 channels like US and Chinese bands, override the said methods.

Reply:
Yes, the phy code appears to not care about any such structure - it just has a certain number of channels, and the mask just has 1 bit per channel number. If or how the channel numbers are split between sub-bands doesn't seem to be relevant to the mask.